two_factor_auth.go

```diff
@@ -0,0 +1,37 @@
+package packngo
+
+const twoFactorAuthAppPath = "/user/otp/app"
+const twoFactorAuthSmsPath = "/user/otp/sms"
+
+// TwoFactorAuthService interface defines available two factor authentication functions
+type TwoFactorAuthService interface {
+	EnableApp() (*Response, error)
+	DisableApp() (*Response, error)
+	EnableSms() (*Response, error)
+	DisableSms() (*Response, error)
+}
+
+// TwoFactorAuthServiceOp implements TwoFactorAuthService
+type TwoFactorAuthServiceOp struct {
+	client *Client
+}
+
+// EnableApp function enables two factor auth using authenticatior app
+func (s *TwoFactorAuthServiceOp) EnableApp() (resp *Response, err error) {
```

Comment:
On HTTP request 

```
=======[REQUEST]=============
POST /user/otp/sms HTTP/1.1
[empty body]
```

I get reply:

```
=======[RESPONSE]============
HTTP/2.0 424 Failed Dependency
{"errors":["Invalid OTP"]}
```


```

Reply:
dtto for the EnableSms

Reply:
.. and indeed, for setting this up in Packet portal, I'd need to supply phone number or current code from Auth App.

Reply:
It is failing for me too.
The phone number should already in your profile so enabling SMS should work. 
According to this https://www.packet.net/developers/api/twofactorauth/ no other parameters should be provided.

Reply:
I have my phone number in my profile but the SMS enabling fails too.

I guess that the 2FA support in API is "work in progress" then, because the phone number or code from auth app are indeed necessary for enabling the respective mode of the 2FA.

In the api-docs JSON there are some more operations, like SMS receive and OTP verify.

Reply:
The 2FA endpoint is used by the Packet UI for managing 2FA. I logged into UI, watched the network requests through the developer mode, identified the behavior for SMS:

Note: Token removed from the following `curl` commands.

### Enable 2FA

1. Specify the SMS phone number:
```
curl -XPATCH https://api.packet.net/user?include=emails -d '{"phone_number":"12063217575"}'
```
2. Send the SMS code:
```
curl -XPOST https://api.packet.net/user/otp/sms/receive
```
3. Validate and enable 2FA:
```
curl -XPOST -H 'x-otp-token: 499510' https://api.packet.net/user/otp/sms
```

### Disable 2FA

1. Send the SMS code:
```
curl -XPOST https://api.packet.net/user/otp/sms/receive
```
2. Validate and disable 2FA:
```
curl -XDELETE -H 'x-otp-token: 499510' https://api.packet.net/user/otp/sms
```

So `packngo` can handle 2FA easily enough. The problem is the feature cannot be tested easily. Any suggestions on how we can handle testing?